File: CORREDOR/Tipos Constructores/Java/Animal.java
package Java;

// Esta clase es abstracta porque NO se puede instanciar directamente.
// Solo sirve como plantilla para otras clases hijas (como Perro).
abstract class Animal{
    String nombre;
    int edad;

    // Constructor CON parámetros
    public Animal(String nombre, int edad) {
        this.nombre = nombre;
        this.edad = edad;
    }

    // Método abstracto (debe ser implementado por las clases hijas)
    abstract void informacion();
}

File: CORREDOR/Tipos Constructores/JS/Animal.js
const AccionesAnimal = require("./AccionesAnimal");

// Constructor CON parámetros 
class Animal {
    constructor(nombre, edad) {
        this.nombre = nombre;
        this.edad = edad;
    }

    informacion(){
        throw new Error("El método 'informacion()' debe ser implementado por la clase hija.");
    }
}
module.exports = Animal;